### PR TITLE
Return 503 from healthhandler when model is not ready

### LINKED ISSUE #799: kubernetes probes using HealthHandler endpoint succeed when models are not ready
/kind bug

**What steps did you take and what happened:**
Used the [HealthHandler endpoint](https://github.com/kubeflow/kfserving/blob/92c5f3c9df662c853beef86bddf7f30ab5d8b996/python/kfserving/kfserving/kfserver.py#L62-L63) to set up a Kubernetes readiness probe. The probe succeeded even though the model was not ready. 

**What did you expect to happen:**
Readiness probe failure when model is not ready. 

**Anything else you would like to add:**
[Kubernetes probe documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) says `Any code greater than or equal to 200 and less than 400 indicates success. Any other code indicates failure`, which seems to imply that the [response contents](https://github.com/kubeflow/kfserving/blob/92c5f3c9df662c853beef86bddf7f30ab5d8b996/python/kfserving/kfserving/kfserver.py#L108-L111) (which include the model's `.ready` property) are ignored. 

**Environment:**

- Istio Version:
- Knative Version:
- KFServing Version:
- Kubeflow version:
- Minikube version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by kolasanichaitanya**
If the model is not marked ready, then don't return 200 from health
check handler, return a 503 instead so it can be used for kubernetes
readiness probe

**What this PR does / why we need it**: Return 503 from healthhandler when model is not ready

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #799 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: python/kfserving/test/test_server.py [hunk 1]
```diff
@@ -65,3 +66,18 @@ async def test_explain(self, http_server_client):
...
+    def app(self):  # pylint: disable=no-self-use
+        model = DummyModel("TestModel")
+        server = kfserving.KFServer()
+        server.register_model(model)
+        return server.create_application()
+
+    async def test_model_not_ready_error(self, http_server_client):
+        with pytest.raises(HTTPClientError) as excinfo:
+            _ = await http_server_client.fetch('/v1/models/TestModel')
+        assert excinfo.value.code == 503
```
Comment (yuzisun) [inline python/kfserving/test/test_server.py:83]: Nit: lint error
```
INFO|2020-05-09T18:52:38|util.py:72| ************* Module test.test_server
INFO|2020-05-09T18:52:38|util.py:72| C: 83, 0: Final newline missing (missing-final-newline)
```

Comment (kolasanichaitanya) [reply python/kfserving/test/test_server.py]: Fixed

**Comment by yuzisun** [comment]
nice fix! thanks for the contribution @kolasanichaitanya !

/lgtm
/approve

**Pull request merged**